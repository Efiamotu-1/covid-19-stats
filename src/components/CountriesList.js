import React from 'react';
import { useSelector } from 'react-redux';
import Countries from './Countries';

const CountriesList = () => {
  const countries = useSelector((state) => state.country.countries);
  const flag = useSelector((state) => state.flags.flag);
  return (
    <div>
      <Countries countries={countries} flag={flag} />
    </div>
  );
};

export default CountriesList;
