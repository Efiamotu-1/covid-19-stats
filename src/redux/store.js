import { configureStore } from '@reduxjs/toolkit';
import countriesSlice from './countries/countriesSlice';
import flagSlice from './countries/flagSlice';
import covidStatsSlice from './statistics/covidStatsSlice';
import statsReducer from './statistics/statsSlice';

const store = configureStore({
  reducer: {
    stats: statsReducer,
    stateStats: covidStatsSlice,
    country: countriesSlice,
    flags: flagSlice,
  },
});

export default store;
