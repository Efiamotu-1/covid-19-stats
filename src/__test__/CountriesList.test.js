import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CountriesList from '../components/CountriesList';
import store from '../redux/store';

describe('Countries List test', () => {
  it('renders the exact same elements', () => {
    const countriesList = render(
      <React.StrictMode>
        <BrowserRouter>
          <Provider store={store}>
            <CountriesList />
          </Provider>
        </BrowserRouter>
      </React.StrictMode>,
    );
    expect(countriesList).toMatchSnapshot();
  });
});
