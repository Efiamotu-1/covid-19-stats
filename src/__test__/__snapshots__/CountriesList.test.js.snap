// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries List test renders the exact same elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex mx-auto w-10/12 justify-center"
        >
          <input
            class="mx-auto w-1/2 my-5 py-3 px-5 rounded-full text-black outline-0"
            placeholder="search for a country"
            type="text"
            value=""
          />
        </div>
        <div
          class="flex-wrap my-5 grid grid-cols-2 md:grid-cols-3 bg-blue-900"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex mx-auto w-10/12 justify-center"
      >
        <input
          class="mx-auto w-1/2 my-5 py-3 px-5 rounded-full text-black outline-0"
          placeholder="search for a country"
          type="text"
          value=""
        />
      </div>
      <div
        class="flex-wrap my-5 grid grid-cols-2 md:grid-cols-3 bg-blue-900"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
